src/regex_impl.cc

```diff
@@ -274,7 +283,19 @@ struct RegexParser
 
         switch (const Codepoint cp = *m_pos)
         {
-            case '.': ++m_pos; return new_node(ParsedRegex::AnyChar);
+            case '.':
+                ++m_pos;
+                if (m_dot_maches_newline)
+                    return new_node(ParsedRegex::AnyChar);
+                else
+                {
+                    CharacterClass c;
```

Comment:
I think this could warrant its own vm op, or at the very least, the character class should be shared, here we'll get a character class for each '.' appearing in the regex.